File: win-prog/ProgWinCS/Images and Bitmaps/CenterImage/CenterImage.cs
//------------------------------------------
// CenterImage.cs © 2001 by Charles Petzold
//------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class CenterImage: PrintableForm
{
     Image image;

     public new static void Main()
     {
          Application.Run(new CenterImage());
     }
     public CenterImage()
     {
          Text = "Center Image";

          image = Image.FromFile("..\\..\\..\\Apollo11FullColor.jpg");
     }
     protected override void DoPage(Graphics grfx, Color clr, int cx, int cy)
     {
          grfx.PageUnit = GraphicsUnit.Pixel;
          grfx.PageScale = 1;

          RectangleF rectf = grfx.VisibleClipBounds;

          float cxImage = grfx.DpiX * image.Width / 
                                             image.HorizontalResolution;
          float cyImage = grfx.DpiY * image.Height / 
                                             image.VerticalResolution;

          grfx.DrawImage(image, (rectf.Width  - cxImage) / 2, 
                                (rectf.Height - cyImage) / 2);
     }
}

File: win-prog/ProgWinVB/Images and Bitmaps/CenterImage/CenterImage.vb
'--------------------------------------------
' CenterImage.vb (c) 2002 by Charles Petzold
'--------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class CenterImage
    Inherits PrintableForm

    Private img As image

    Shared Shadows Sub Main()
        Application.Run(new CenterImage())
    End Sub

    Sub New()
        Text = "Center Image"
        img = Image.FromFile("..\..\Apollo11FullColor.jpg")
    End Sub

    Protected Overrides Sub DoPage(ByVal grfx As Graphics, _
            ByVal clr As Color, ByVal cx As Integer, ByVal cy As Integer)
        grfx.PageUnit = GraphicsUnit.Pixel
        grfx.PageScale = 1
        Dim rectf As RectangleF = grfx.VisibleClipBounds
        Dim cxImage As Single = grfx.DpiX * img.Width / _
                                            img.HorizontalResolution
        Dim cyImage As Single = grfx.DpiY * img.Height / _
                                            img.VerticalResolution
        grfx.DrawImage(img, (rectf.Width - cxImage) / 2, _
                            (rectf.Height - cyImage) / 2)
    End Sub
End Class
